src/capymoa/anomaly/online_isolation_forest.py

```diff
@@ -0,0 +1,341 @@
+    ...     learner.train(instance)
+    >>> auc = evaluator.auc()
+    >>> print(f"AUC: {auc:.2f}")
+    AUC: 0.52
+    """
+    def __init__(self, schema: Schema = None, random_seed: int = 1, num_trees: int = 32, max_leaf_samples: int = 32,
+                 type: str = 'adaptive', subsample: float = 1.0, window_size: int = 2048, branching_factor: int = 2,
+                 metric: str = 'axisparallel', n_jobs: int = 1):
+        """Construct an Online Isolation Forest anomaly detector
+
+        :param schema: The schema of the stream. If not provided, it will be inferred from the data.
+        :param random_seed: Random seed for reproducibility.
+        :param num_trees: Number of trees in the ensemble.
+        :param window_size: The size of the window for each tree.
+        :param branching_factor: Branching factor of each tree.
+        :param max_leaf_samples: Maximum number of samples per leaf. When this number is reached, a split is performed.
+        :param type: Type of split performed. If "adaptive", the max_leaf_samples grows with tree depth.
+        :param subsample: Probability of learning a new sample in each tree.
+        :param n_jobs: Number of parallel jobs.
+        """
+        super().__init__(schema=schema, random_seed=random_seed)
+        self.num_trees: int = num_trees
+        self.window_size: int = window_size
+        self.branching_factor: int = branching_factor
+        self.max_leaf_samples: int = max_leaf_samples
+        self.type: str = type
+        self.subsample: float = subsample
+        self.trees: list[OnlineIsolationTree] = []
+        self.data_window: list[ndarray] = []
+        self.data_size: int = 0
+        self.normalization_factor: float = 0
+        self.metric: str = metric
+        self.n_jobs: int = cpu_count() if n_jobs == -1 else min(n_jobs, cpu_count())
+        self.trees: list[OnlineIsolationTree] = [OnlineIsolationTree(max_leaf_samples=max_leaf_samples,
+                                                                     type=type,
+                                                                     subsample=self.subsample,
+                                                                     branching_factor=self.branching_factor,
+                                                                     data_size=self.data_size,
+                                                                     metric=self.metric,
+                                                                     random_seed=self.random_seed) for _ in range(self.num_trees)]
+
+    def train(self, instance: Instance):
+        data: ndarray = instance.x.reshape((1, -1))
+        self.learn_batch(data)
+        return
+
+    def predict(self, instance: Instance) -> Optional[LabelIndex]:
+        pass
+
+    def score_instance(self, instance: Instance) -> AnomalyScore:
+        data: ndarray = instance.x.reshape((1, -1))
+        return self.score_batch(data)
```

Comment:
This should convert this to a double/float before returning it. The implicit conversion will be an error in a future version of numpy
```
Conversion of an array with ndim > 0 to a scalar is deprecated, and will error in future.
```